src/Symfony/Component/Env/Dotenv.php

```diff
@@ -0,0 +1,380 @@
+        if (!$singleQuoted) {
+            $value = $this->resolveVariables($value);
+            $value = $this->resolveCommands($value);
+        }
+
+        if ($notQuoted && $currentValue == $value && preg_match('/\s+/', $value)) {
+            throw new FormatException('A value containing spaces must be surrounded by quotes', $this->createFormatExceptionContext());
+        }
+
+        return $value;
+    }
+
+    private function skipWhitespace()
+    {
+        $this->cursor += strspn($this->data, ' ', $this->cursor);
+    }
+
+    private function skipEmptyLines()
+    {
+        if (preg_match('/(\n+|^#[^\n]*(\n*|$))+/Asm', $this->data, $match, null, $this->cursor)) {
+            $this->moveCursor($match[0]);
+        }
+    }
+
+    private function skipComment()
+    {
+        if (preg_match('/ *#[^\n]*(\n*|$)/Asm', $this->data, $match, null, $this->cursor)) {
+            $this->moveCursor($match[0]);
+        }
+    }
+
+    private function resolveCommands($value)
+    {
+        if (false === strpos($value, '$')) {
+            return $value;
+        }
+
+        $regex = '/
+            (\\\\)?               # escaped with a backslash?
+            \$
+            (?<cmd>
+                \(                # require opening parenthesis
+                ([^()]|\g<cmd>)+  # allow any number of non-parens, or balanced parens (by nesting the <cmd> expression recursively)
+                \)                # require closing paren
+            )
+        /x';
+
+        $env = array_replace($_ENV, $_SERVER, $this->values);
+        $path = $this->path;
+        $lineno = $this->lineno;
+
+        return preg_replace_callback($regex, function ($matches) use ($env, $path, $lineno) {
+            if ('\\' === $matches[1]) {
+                return substr($matches[0], 1);
+            }
+
+            if (!class_exists('Symfony\Component\Process\Process')) {
+                throw new \LogicException('Resolving commands requires the Symfony Process component.');
+            }
+
+            $process = new Process('echo '.$matches[0], null, $env);
+            try {
+                $process->mustRun();
+            } catch (ProcessException $e) {
+                throw new FormatException(sprintf('Issue expanding a command (%s)', $process->getErrorOutput()), $this->createFormatExceptionContext());
+            }
+
+            return preg_replace('/[\r\n]+$/', '', $process->getOutput());
+        }, $value);
+    }
+
+    private function resolveVariables($value)
+    {
+        if (false === strpos($value, '$')) {
+            return $value;
+        }
+
+        $regex = '/
+            (\\\\)?                    # escaped with a backslash?
+            \$
+            (?!\()                     # no opening parenthesis
+            (\{)?                      # optional brace
+            ('.self::VARNAME_REGEX.')  # var name
+            (\})?                      # optional closing brace
+        /xi';
```

Comment:
The regex is evaluated in case-insensitive mode, while it's case sensitive in the `lexVarname()` method.